### PR TITLE
Feature/20629 occ group commands

### PR DESCRIPTION
**Pull request by edwardcrompton**
Adding a console command for creating groups.

> occ group:add <newgroup>

### CODE REVIEW

**Comment by karlitschek** [comment]
@blizzz @PVince81 what do you think?

File: core/command/group/add.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+class Add extends Command {
+	/** @var \OCP\IUserManager */
+	protected $userManager;
+
+	/** @var \OCP\IGroupManager */
+	protected $groupManager;
+
+	/**
+	 */
+	public function __construct(IGroupManager $groupManager) {
```
Comment (DeepDiver1975) [inline core/command/group/add.php]: IGroupManager is undefined

File: core/command/group/add.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?php
+/**
```
Comment (DeepDiver1975) [inline core/command/group/add.php]: please copy over our standard header from another file

File: core/command/group/add.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		$group = $this->groupManager->get($groupName);
+		if (!$group) {
+			$this->groupManager->createGroup($groupName);
+			$group = $this->groupManager->get($groupName);
+			$output->writeln('Created group "' . $group->getGID() . '"');
+		} else {
+			$output->writeln('<error>The group already exists</error>');
+			return 1;
+		}
+  }
```
Comment (DeepDiver1975) [inline core/command/group/add.php]: indent

**Comment by DeepDiver1975** [comment]
Please squash your commits

**Comment by DeepDiver1975** [comment]
@edwardcrompton Thanks a lot for your contribution!
Contributions to the core repo require a signed contributors agreement http://owncloud.org/about/contributor-agreement/

Alternatively you can add a comment here where you state that this contribution is MIT licensed.

Some more details about out pull request workflow can be found here: http://owncloud.org/code-reviews-on-github/

File: core/command/group/add.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+				InputArgument::REQUIRED,
+				'The name of the group to create'
+			);
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$groupName = $input->getArgument('group');
+		$group = $this->groupManager->get($groupName);
+		if (!$group) {
+			$this->groupManager->createGroup($groupName);
```
Comment (nickvergessen) [inline core/command/group/add.php]: Should check the return here.
If the return is false or null, it failed.

**Comment by blizzz** [comment]
@edwardcrompton thanks for your contribution! To get this into ownCloud, could you check back with the things @DeepDiver1975 and @nickvergessen commented on?

**Comment by DeepDiver1975** [comment]
Close due to inactivity

**Pull request closed without merging**